### PR TITLE
nodejs: update to 16.9.1

### PR DESCRIPTION
**Pull request by Dko1905**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [X] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

This pr includes #33036.

### CODE REVIEW

**Comment by Dko1905** [comment]
# Errors in build

The first two errors are timeout errors, no idea what went wrong, or how to fix them.
```
2021-09-20T18:18:34.5672091Z [00:58|%  14|+ 502|-   0]: release test-cluster-primary-kill
2021-09-20T18:18:34.5674805Z                                                             
2021-09-20T18:18:34.5676819Z === release test-cluster-primary-error ===
2021-09-20T18:18:34.5678982Z Path: parallel/test-cluster-primary-error
2021-09-20T18:18:34.5681684Z Command: out/Release/node /builddir/node-v16.9.1/test/parallel/test-cluster-primary-error.js
2021-09-20T18:18:34.5694440Z --- TIMEOUT ---
2021-09-20T18:18:34.5695159Z 
2021-09-20T18:18:34.5695706Z                                                             
2021-09-20T18:18:34.6550606Z [02:58|%  14|+ 502|-   1]: release test-cluster-process-disconnect
2021-09-20T18:18:34.6553094Z                                                                   
2021-09-20T18:18:34.6554996Z === release test-cluster-primary-kill ===
2021-09-20T18:18:34.6556037Z Path: parallel/test-cluster-primary-kill
2021-09-20T18:18:34.6557765Z Command: out/Release/node /builddir/node-v16.9.1/test/parallel/test-cluster-primary-kill.js
2021-09-20T18:18:34.6558797Z --- TIMEOUT ---
2021-09-20T18:18:34.6559093Z 
2021-09-20T18:18:34.6559483Z
```

And the last error is a service lookup, but it fails because `/etc/services` isn't a file in the chroot, as far as I know.
```
2021-09-20T18:19:39.4329419Z === release test-dns-lookupService-promises ===
2021-09-20T18:19:39.4330960Z Path: parallel/test-dns-lookupService-promises
2021-09-20T18:19:39.4332021Z node:internal/process/promises:246
2021-09-20T18:19:39.4333051Z           triggerUncaughtException(err, true /* fromPromise */);
2021-09-20T18:19:39.4333938Z           ^
2021-09-20T18:19:39.4334358Z 
2021-09-20T18:19:39.4335200Z AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
2021-09-20T18:19:39.4336313Z 
2021-09-20T18:19:39.4337184Z '22' !== 'ssh'
2021-09-20T18:19:39.4338076Z 
2021-09-20T18:19:39.4339460Z     at /builddir/node-v16.9.1/test/parallel/test-dns-lookupService-promises.js:9:10
2021-09-20T18:19:39.4340971Z     at /builddir/node-v16.9.1/test/common/index.js:394:15 {
2021-09-20T18:19:39.4341847Z   generatedMessage: true,
2021-09-20T18:19:39.4342985Z   code: 'ERR_ASSERTION',
2021-09-20T18:19:39.4343842Z   actual: '22',
2021-09-20T18:19:39.4344632Z   expected: 'ssh',
2021-09-20T18:19:39.4345532Z   operator: 'strictEqual'
2021-09-20T18:19:39.4346175Z }
2021-09-20T18:19:39.4347592Z Command: out/Release/node /builddir/node-v16.9.1/test/parallel/test-dns-lookupService-promises.js
2021-09-20T18:19:39.4349448Z
```

Now I just need a way of ignoring them, any help would be nice.

**Comment by Dko1905** [comment]
I am now ignoring the `TIMEOUT` errors, I'll first try running it on my computer, and if it works, I'll push.

**Comment by Dko1905** [comment]
The tests now run fine on my computer.

File: srcpkgs/nodejs/template [hunk 1]
```diff
@@ -50,6 +50,7 @@ esac
 
 CFLAGS="-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64"
 CXXFLAGS="-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64"
+COV_SKIP_TESTS="core_line_numbers.js,testFinalizer.js,test_function/test.js,test-cluster-primary-error.js,test-cluster-primary-kill.js"
```
Comment (Johnnynator) [inline srcpkgs/nodejs/template:53]: Keep custom variables in a local function scope if possible (xbps-src doesn't always clean them up properly otherwise)

**Comment by Dko1905** [comment]
The `COV_SKIP_TESTS` variable does not seem to work, the skipped tests are still run, and I can see in the logs that the `--skip-tests=` is empty in the Makefile.
<br/>
Snippet from log:
```
2021-09-21T14:45:17.1874570Z if [ ! -r node ] || [ ! -L node ]; then \
2021-09-21T14:45:17.1875203Z   ln -fs out/Release/node node; fi
2021-09-21T14:45:17.1903062Z /usr/bin/python3.9 tools/test.py -J --mode=release \
2021-09-21T14:45:17.1903517Z 	 \
2021-09-21T14:45:17.1903969Z 	--skip-tests= \
2021-09-21T14:45:17.1904315Z 	default \
2021-09-21T14:45:17.1904850Z 	addons js-native-api node-api
```

And I think you're right about llhttp breaking, if we merge 6.x.

File: srcpkgs/nodejs/template (hunk 1 above)
Comment (Dko1905) [reply srcpkgs/nodejs/template:53]: outdated

Comment (Dko1905) [reply srcpkgs/nodejs/template:53]: And does not work.

**Comment by Dko1905** [comment]
I can see that I am wrong, the `COV_SKIP_TESTS` only applies when calculating test coverage. The correct variable would be `CI_SKIP_TESTS`.

**Comment by Dko1905** [comment]
Thank you, I look forward to future contributions.

**Pull request merged**